state/protocol/events/distributor.go

```diff
@@ -71,3 +71,11 @@ func (d *Distributor) EpochEmergencyFallbackTriggered() {
 		sub.EpochEmergencyFallbackTriggered()
 	}
 }
+
+func (d *Distributor) ClusterIdsUpdated(list flow.ChainIDList) {
```

Comment:
I would like to decouple this new event from the others, because it is originating from a different component. All existing events here are published from the Protocol State, and this new event is published from the `epochmgr` engine. 

We recently [refactored the HotStuff pubsub system in a similar way](https://github.com/onflow/flow-go/blob/2afe79e0d7360acffc30d4bfb1301cbba7a86e44/consensus/hotstuff/notifications/pubsub/distributor.go#L10-L14), where the event interfaces and distributor implementations are split into pieces depending on the component which publishes the events.

I think we should apply that design here as well.

Reply:
https://github.com/onflow/flow-go/pull/4189/commits/ee79571ad40e8aea5207a2ada5563d2e34aab805